tiled/utils.py

```diff
@@ -8,6 +8,7 @@
 import os
 import sys
 import threading
+import numpy
```

Comment:
This approach, falling back to `default`, is great.

My one note is that we try to avoid importing the "heavy" libraries like `numpy` unless/until they are used. This gets us two things:

1. For minimalist applications that only require access to basic metadata—say, a process that counts the number of scans done so far today—the environment doesn't need to install `numpy`/`pandas`/`xarray`.
2. The top-level import `import tiled` runs quickly, and tools like the CLI `tiled profile list` or `tiled serve ...` aren't slowed down by waiting for these large packages to import (and then not even be used).

So, the goal is to avoid importing numpy unless it is has already been imported somewhere else in the program (at which point `import numpy` is basically free). Here's what we can do. If numpy hasn't been imported yet, we can be sure that there are no instances of `numpy.str_` or `numpy.bytes_`. Therefore, we can protect the import like this:

```py
    def default(content):
        # If numpy has not been imported, we can be sure that there are no
        # instances of numpy objects, and we don't want to pay for the numpy
        # import / dependency if it's not otherwise used.
        if "numpy" in sys.modules:
            import numpy

            if isinstance(content[0], (numpy.str_, numpy.bytes_)):
                return content.tolist()
        return super().default(content)
```